app/models/user.rb

```diff
@@ -0,0 +1,35 @@
+class User < ActiveRecord::Base
+  belongs_to :role
+  has_many :reviews
+  has_many :votes
+  has_many :ratings
+  has_many :comments
+
+  validates :email, :role_id, presence: true
+  validates :email, uniqueness: true
+  validates :role_id, inclusion: {in: [1,2,3]}
```

Comment:
Is 1,2,3 a "magic number" analog? Think about making it more explicit. Let's discuss.